src/log_sifter.py

```diff
@@ -0,0 +1,161 @@
+""" Module for completing end to end parsing of a log """
+
+import json
+
+import constants
+from log_parser import LogParser
+from event_parser import LogEventParser
+from group_parser import EventGroupParser 
+from symptom_burst import SymptomBurst
+from symptom import Symptom
+from signals import DetectedSignal
+from log_event_group import LogEventGroup
+from log_event import LogEvent
+
+class LogSifter:
+    """ Object for implementing the high level flow of parsing the log
+
+        Attributes
+        ----------
+        log_file_path: string relative file path to the log
+        event_rules: list of EventRule objects to be applied
+        group_rules: list of EventGroupRules to be applied
+        criterias: list of Criteria objects to be applied
+        flags: Objcet for storing the build flags from the CLI """
+
+    def __init__(self, log_file_path, event_rules, group_rules, criterias, flags):
+        self.log_file_path = log_file_path
+        self.event_rules = event_rules
+        self.group_rules = group_rules
+        self.criterias = criterias
+        self.flags = flags
+
+    def sift_log(self):
+        """ Implements the logical flow of the multiple parsing stages. """
+
+        # Parse the raw log
+        log_parser = LogParser(self.log_file_path, self.event_rules)
+        log_parser.parse_log()
+        log_events = log_parser.log_events_found
+
+        # Parse the log events
+        event_parser = LogEventParser(log_events, self.group_rules)
+        event_parser.parse_log_events()
+        event_groups = event_parser.event_groups_found
+
+        collect_statistics = self.flags.collect_statistics
+
+        # Parse the event groups
+        group_parser = EventGroupParser(event_groups, self.criterias,
+                                        collect_statistics)
+        group_parser.parse_event_groups()
+        statistics = group_parser.statistics_summaries
+
+        # Construct the output
+        if self.flags.json_output:
+            self.output_to_json(group_parser, statistics)
+        else:
+            self.output_to_terminal(group_parser, statistics)
+
+    def output_to_terminal(self, group_parser, statistics):
+        """ Outputs a listing of each symptom burst and its details to
+            the terminal depending on the level of verbosity selected. """
+
+        burst_tags = list(group_parser.burst_dict.keys())
+        burst_dict = group_parser.burst_dict
+        burst_list = group_parser.bursts
+
+        symptom_list = group_parser.symptoms_found
+
+        if self.flags.verbosity == constants.VerbosityLevel.HIGH:
+            for symptom in symptom_list:
```

Comment:
I don't really understand the point of this for loop. If this is just a TODO to be addressed later, then you could probably just put the TODO at the top level rather than making a do-nothing for loop.

Reply:
oof! That was missing implementation altogether, I'll fix that.